agent/agent.go

```diff
@@ -17,13 +17,15 @@ import (
 
 // Agent runs a set of plugins.
 type Agent struct {
-	Config *config.Config
+	Config  *config.Config
+	RunOnce bool
 }
 
 // NewAgent returns an Agent for the given Config.
-func NewAgent(config *config.Config) (*Agent, error) {
+func NewAgent(config *config.Config, runOnce bool) (*Agent, error) {
```

Comment:
The implementation of this isn't something I would accept, adding booleans as trailing parameters is not allowed and it pushes a lot of new complexity into the agent.

The interface to this should be a new function, similar to what was done with the Test function.

Implementation might be shared, but it ought to guarantee a single Gather, single Aggregation, and as many writes are are required to send it all.  So, I'm not sure if sharing much would work out.